### ISSUE TITLE
Invalid Key triggers Segfault

### ISSUE THREAD

**Issue by dominiek** [coreos/kube-aws#234]
In previous versions this error occurred when the `kmsKeyArn` was invalid:

```
Validating UserData...
Error: failed to compress TLS assets: NotFoundException: Key 'XXXX' does not exist
	status code: 400, request id: 5f985cad-d768-11e6-b25e-c3e26016f5e9
```

But in latest versions including `v0.9.3-rc.2` the following happens:

```
Validating UserData...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x964f8]

goroutine 1 [running]:
panic(0x617cc0, 0xc4200120a0)
	/usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
github.com/coreos/kube-aws/config.(*EncryptedTLSAssets).Compact(0x0, 0xb, 0xc420124750, 0x9)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/tls_config.go:375 +0xc8
github.com/coreos/kube-aws/config.ReadOrCreateCompactTLSAssets(0x6fe69d, 0xb, 0xc420124750, 0x9, 0x0, 0x0, 0xc420011090, 0x4a, 0xc4201246d8, 0x8, ...)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/tls_config.go:437 +0x6a
github.com/coreos/kube-aws/config.Cluster.stackConfig(0xc4200d3400, 0x18, 0x6fd0a4, 0x9, 0x0, 0x0, 0x0, 0x0, 0xc4201246d8, 0x8, ...)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/config.go:618 +0x186
github.com/coreos/kube-aws/config.Cluster.ValidateUserData(0xc4200d3400, 0x18, 0x6fd0a4, 0x9, 0x0, 0x0, 0x0, 0x0, 0xc4201246d8, 0x8, ...)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/config.go:636 +0x77
github.com/coreos/kube-aws/cmd.runCmdValidate(0xaea7c0, 0xc4200d32e0, 0x0, 0x2, 0x0, 0x0)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/cmd/validate.go:50 +0x19d
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).execute(0xaea7c0, 0xc4200d31c0, 0x2, 0x2, 0xaea7c0, 0xc4200d31c0)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:632 +0x23e
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xae9f40, 0x0, 0x0, 0x0)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:722 +0x367
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).Execute(0xae9f40, 0xc420145f50, 0xc4200001a0)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:681 +0x2b
main.main()
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/main.go:9 +0x2d
```

**Comment by redbaron** [comment]
fixed in master

**Comment by mumoshu** [comment]
@dominiek Thanks for reporting the issue!
Sorry for the inconvenience but the fix in included in v0.9.3-rc.3.
Could mind retrying with it and reopen if the issue still persists?
